src/bindings/c/ov/include/c_api/ov_c_api.h

```diff
@@ -0,0 +1,1122 @@
+ * in this case, std::map contains multiple entries, each per device.
+ * @param core A pointer to the ov_core_t instance.
+ * @param device_name Device name to identify a plugin.
+ * @param versions A pointer to versions corresponding to device_name.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e) ov_core_get_versions(
+                                        const ov_core_t* core,
+                                        const char* device_name,
+                                        ov_core_version_list_t *versions);
+
+/**
+ * @brief Releases memory occupied by ov_core_version_list_t.
+ * @param vers A pointer to the ie_core_versions to free memory.
+ */
+OPENVINO_C_API(void) ov_core_versions_free(ov_core_version_list_t *versions);
+
+/**
+ * @brief Get the outputs of ov_model_t.
+ * @param model A pointer to the ov_model_t.
+ * @param output_nodes A pointer to the ov_output_nodes.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e) ov_model_get_outputs(const ov_model_t* model, ov_output_node_list_t *output_nodes);
+
+/**
+ * @brief Get the outputs of ov_model_t.
+ * @param model A pointer to the ov_model_t.
+ * @param input_nodes A pointer to the ov_input_nodes.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e) ov_model_get_inputs(const ov_model_t* model, ov_output_node_list_t *input_nodes);
+
+/**
+ * @brief Get the tensor name of ov_output_node.
+ * @param nodes A pointer to the ov_output_node_list_t.
+ * @param idx Index of the input tensor
+ * @param tensor_name A pointer to the tensor name.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e) ov_node_get_tensor_name(ov_output_node_list_t *nodes, size_t idx, char** tensor_name);
+
+/**
+ * @brief Get the tensor shape of ov_output_node.
+ * @param nodes A pointer to the ov_output_node_list_t.
+ * @param idx Index of the input tensor
+ * @param tensor_shape tensor shape.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e) ov_node_get_tensor_shape(ov_output_node_list_t *nodes, size_t idx, ov_shape_t* tensor_shape);
+
+/**
+ * @brief Get the tensor type of ov_output_node.
+ * @param nodes A pointer to the ov_output_node_list_t.
+ * @param idx Index of the input tensor
+ * @param tensor_type tensor type.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e) ov_node_get_tensor_type(ov_output_node_list_t *nodes, size_t idx, ov_element_type_e *tensor_type);
+
+/**
+ * @brief Get the outputs of ov_model_t.
+ * @param model A pointer to the ov_model_t.
+ * @param tensor_name input tensor name (char *).
+ * @param input_node A pointer to the ov_output_node_t.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e) ov_model_get_input_by_name(const ov_model_t* model,
```

Comment:
Too simple name will confuse user? I'm not sure.

Reply:
let's move methods to work with ov_model to a dedicated file. I suppose such API is not very common since most frequent usage is to compile model without modification.